JSDemos/Demos/DataGrid/CustomNewRecordPosition/React/App.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import DataGrid, {
+  Column, Editing, ValidationRule, Button, Toolbar, Item, Scrolling,
+} from 'devextreme-react/data-grid';
+import SelectBox from 'devextreme-react/select-box';
+import Guid from 'devextreme/core/guid';
+import { dataSource } from './data.js';
+
+const newRowPositionOptions = ['first', 'last', 'pageTop', 'pageBottom', 'viewportTop', 'viewportBottom'];
+const scrollingModeOptions = ['standard', 'virtual'];
+
+function App() {
+  const [newRowPosition, setNewRowPosition] = React.useState('viewportTop');
+  const [scrollingMode, setScrollingMode] = React.useState('standard');
+  const [changes, setChanges] = React.useState([]);
+  const [editRowKey, setEditRowKey] = React.useState(null);
+
+  const onOptionChanged = React.useCallback((e) => {
+    if (e.fullName === 'editing.changes') {
+      setChanges(e.value);
+    } else if (e.fullName === 'editing.editRowKey') {
+      setEditRowKey(e.value);
```

Comment:
Use editing.onEditRowKeyChange and editing.onChangesChange props instead of onOptionChanged. See the https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/EditStateManagement/React/Light/ demo as example.